types/react-relay/index.d.ts

```diff
@@ -87,9 +90,9 @@ export class QueryRenderer extends ReactRelayQueryRenderer {}
 // createFragmentContainer
 // ~~~~~~~~~~~~~~~~~~~~~
 export function createFragmentContainer<T>(
-    Component: ReactBaseComponent<T>,
-    fragmentSpec: GraphQLTaggedNode | GeneratedNodeMap
-): ReactBaseComponent<T>;
+    Component: React.ComponentType<T>,
+    fragmentSpec: RelayRuntimeTypes.GraphQLTaggedNode | GeneratedNodeMap
+): React.ComponentType<RemoveRelayProp<T> & { componentRef?: any }>;
```

Comment:
Simply added `componentRef` here and as `any` like the Flow types in upstream Relay does as well. It would be great if we can improve this in the future to _actually_ be typed as the input component class, but for now wanted to get this out of the door first.

Reply:
Having messed around with this, it seems undoable until TS 2.8 - so any should be good for now (we could type it as: `(ref: any) => void` if we wanted to get a bit closer).

Reply:
Ah yes, good point! Will type it as a callback 👍